### ISSUE TITLE
esp-mqtt does not compile with SSL enabled (IDFGH-2620)

### ISSUE THREAD

**Issue by eespi** [espressif/esp-mqtt#149]
Compile error after pulling master
```
/media/paolo/Data/working/FW/esp32/esp-idf/components/mqtt/esp-mqtt/mqtt_client.c: In function 'esp_mqtt_set_ssl_transport_properties':
/media/paolo/Data/working/FW/esp32/esp-idf/components/mqtt/esp-mqtt/mqtt_client.c:236:9: error: label '_mqtt_set_config_failed' used but not defined
goto _mqtt_set_config_failed;
```

**Comment by david-cermak** [comment]
@eespi Thanks for the quick report.

The issue is that it doesn't compile with older IDFs, posting a quick fix:
[0001-fix-compilation-issue-on-v4.0.patch.txt](https://github.com/espressif/esp-mqtt/files/4140132/0001-fix-compilation-issue-on-v4.0.patch.txt)
before fixing on master.

Will add a CI check to compile the library agains all major releases from v3.2

**Comment by eespi** [comment]
thanks, it works now
